src/main/java/duke/data/Storage.java

```diff
@@ -0,0 +1,85 @@
+package duke.data;
+
+import duke.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private final File data;
+    private final String[] dataReadWriteErrorMsg = {
```

Comment:
Suggest to replace this array with individual variables, or replace magic number with the appropriately-named constant.